Sources/OVP/Services/OVPBaseEntryService.swift

```diff
@@ -15,14 +15,19 @@ import KalturaNetKit
 
 class OVPBaseEntryService {
 
-    internal static func list(baseURL: String, ks: String,entryID: String) -> KalturaRequestBuilder? {
+    internal static func list(baseURL: String, ks: String, entryID: String?, referenceId: String?) -> KalturaRequestBuilder? {
         
         if let request: KalturaRequestBuilder = KalturaRequestBuilder(url: baseURL, service: "baseEntry", action: "list") {
             let responseProfile = ["fields": "mediaType,dataUrl,id,name,duration,msDuration,flavorParamsIds,tags,dvrStatus",
                                    "type": 1] as [String: Any]
             request.setBody(key: "ks", value: JSON(ks))
                 .setBody(key: "responseProfile", value: JSON(responseProfile))
-                .setBody(key: "filter:redirectFromEntryId", value: JSON(entryID))
+                
+            if let entryID = entryID {
+                request.setBody(key: "filter:redirectFromEntryId", value: JSON(entryID))
+            } else if let referenceId = referenceId {
+                request.setBody(key: "filter:referenceIdEqual", value: JSON(referenceId))
+            }
```

Comment:
Can there be an entryID and referenceId?

Reply:
Same logic we have on [Android](https://github.com/kaltura/playkit-android-providers/blob/df96831ee5edfbe23a39b7996c526d6afc12fef8/mediaproviders/src/main/java/com/kaltura/playkit/providers/api/ovp/services/BaseEntryService.java#L61) and [WEB](https://github.com/kaltura/playkit-js-providers/blob/dd68082fb32be21e26cc92439c1d9628b581c8a7/src/k-provider/ovp/services/base-entry-service.js#L68)